File: tests/python/src/basicrequester.py
import sys
import time
import cameopy

this = cameopy.This
this.init(sys.argv)

useProxy = False if len(sys.argv) < 2 else (sys.argv[1] == "true")

options = 0
if useProxy:
    options |= cameopy.USE_PROXY

starter = this.connectToStarter(options)

app = starter.getApp()

# Create a requester.
requester = cameopy.coms.Requester.create(app, "responder")
requester.init()

this.setRunning()

# Send 10 requests.
R = 10
for i in range(R):
    # Send and wait for the result.
    requester.send("test")

    response = requester.receiveString()

    if response:
        print("Received",  response)

    time.sleep(0.1)

starter.terminate()

print("Finished the application")

File: tests/cpp/src/BasicRequester.cpp
/*
 * CAMEO
 *
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under BSD 3-Clause and GPL-v3 as described in license files.
 * You may not use this work except in compliance with the Licences.
 *
 */

#include <cameo/api/cameo.h>
#include <iostream>

using namespace std;
using namespace cameo;

int main(int argc, char *argv[]) {

	This::init(argc, argv);

	int options = 0;
	bool useProxy = false;
	if (argc > 2) {
		useProxy = (string(argv[1]) == "true");
	}
	if (useProxy) {
		options |= option::USE_PROXY;
	}

	std::unique_ptr<ServerAndApp> starter = This::connectToStarter(options);

	cout << "Connected to starter" << endl;

	// Create a requester.
	unique_ptr<coms::Requester> requester = coms::Requester::create(starter->getApp(), "responder");
	requester->init();

	cout << "Created requester" << endl;

	This::setRunning();

	// Send 10 requests.
	int R = 10;
	for (int i = 0; i < R; ++i) {
		// Send and wait for the result.
		requester->send("test");

		std::optional<std::string> response = requester->receive();

		if (response.has_value()) {
			cout << "Received " << response.value() << endl;
		}
	}

	cout << "Finished the application" << endl;

	return 0;
}